db/system_keyspace.cc

```diff
@@ -2341,6 +2331,9 @@ future<std::set<sstring>> system_keyspace::load_local_enabled_features() {
 future<> system_keyspace::save_local_enabled_features(std::set<sstring> features) {
     auto features_str = fmt::to_string(fmt::join(features, ","));
     co_await set_scylla_local_param(gms::feature_service::ENABLED_FEATURES_KEY, features_str);
+    // Flush the table so that the value is available on boot before commitlog replay.
+    // database::maybe_init_schema_commitlog() depends on it.
+    co_await force_blocking_flush(SCYLLA_LOCAL);
 }
```

Comment:
The `save_local_enabled_features` function is called in two places: in `persistent_feature_enabler::enable_features()`, but also in `storage_service::topology_state_load()`.

In the first case, the flush is necessary because it is related to the legacy features-in-gossip path: we call it after enabling features based on gossip, and we use the features saved in `system.scylla_local` for the feature check in startup.

The second case is related to features-in-raft; we update `system.scylla_local` based on the group 0 state only for legacy purposes, like with `system.peers` etc. In that case you will most certainly want to avoid flushing as `system.scylla_local` is no longer a source of truth for the features and we don't use it in the feature check on startup.

Reply:
done, thanks